internal/metrics/metrics.go

```diff
@@ -91,15 +121,16 @@ func (metrics *Metrics) Reset() {
 }
 
 func (metrics *Metrics) RecordSetupResult(name string, result ResultType, nanoseconds int64) {
-	metrics.Setup.WithLabelValues(name, result.String()).Observe(float64(nanoseconds))
+	labels := append([]string{name, result.String()}, getMetricsValuesFromEnrichment()...)
+	metrics.Setup.WithLabelValues(labels...).Observe(float64(nanoseconds))
 }
 
 func (metrics *Metrics) RecordIterationResult(name string, result ResultType, nanoseconds int64) {
 	if !metrics.IterationMetricsEnabled {
 		return
 	}
-
-	metrics.Iteration.WithLabelValues(name, IterationStage, result.String()).Observe(float64(nanoseconds))
+	labels := append([]string{name, IterationStage, result.String()}, getMetricsValuesFromEnrichment()...)
```

Comment:
this will be called on each iteration and it's static. Ideally we should cache it instead of parsing the env var and building it up each time.